File: 0238 product-of-array-except-self/solution.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int[] pr = new int[nums.Length];
        pr[0] = 1;
        for(int i=1;i<nums.Length;i++) {
            pr[i] = pr[i-1]*nums[i-1];
        }
        int s = 1;
        for(int i = pr.Length-1;i>=0;i--) {
            pr[i]*=s;
            s*=nums[i];
        }
        return pr;
    }
}

File: 0238 product-of-array-except-self/solution.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        arr = [1] * (n+1)
        for i in range(n):
            arr[i+1] = arr[i]*nums[i]
        suf = 1
        for i in range(n-1, -1, -1):
            arr[i]=arr[i]*suf
            suf*=nums[i]
        return arr[:n]
